AmplifyTests/CoreTests/Operations/FastOperation.swift

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+@testable import Amplify
+
+public class FastOperationRequest: AmplifyOperationRequest {
+    public let options: [AnyHashable : Any]
+    public let numbers: [Int]
+
+    public init(options: [AnyHashable : Any] = [:], numbers: [Int]) {
+        self.options = options
+        self.numbers = numbers
+    }
+}
+
+public struct FastOperationSuccess {
+    public let value: Int
+
+    public init(value: Int) {
+        self.value = value
+    }
+}
+
+public enum FastOperationError: AmplifyError {
+    case unknown(ErrorDescription, Error? = nil)
+
+    public var errorDescription: ErrorDescription {
+        switch self {
+        case .unknown(let errorDescription, _):
+            return "Unexpected error occurred with message: \(errorDescription)"
+        }
+    }
+
+    public var recoverySuggestion: RecoverySuggestion {
+        switch self {
+        case .unknown:
+            return "No suggestion"
+        }
+    }
+
+    public var underlyingError: Error? {
+        switch self {
+        case .unknown(_, let error):
+            return error
+        }
+    }
+
+    public init(
+        errorDescription: ErrorDescription = "An unknown error occurred",
+        recoverySuggestion: RecoverySuggestion = "(Ignored)",
+        error: Error
+    ) {
+        if let error = error as? Self {
+            self = error
+        } else {
+            self = .unknown(errorDescription, error)
+        }
+    }
+}
+
+public typealias FastOperationResult = Result<FastOperationSuccess, FastOperationError>
+public typealias FastOperationResultListener = (FastOperationResult) -> Void
+
+public class FastOperation: AmplifyOperation<FastOperationRequest, FastOperationSuccess, FastOperationError> {
+    public typealias TaskAdapter = AmplifyOperation<Request, Success, Failure>
```

Comment:
should this be typealias for `AmplifyOperationTaskAdapter` ?

Reply:
This is the mechanism which bridges to `AmplifyTask` by passing along the generic types of the Operation. When the instance of the adapter is created it passes in the actual `Operation`. 

```swift
    init(operation: AmplifyOperation<Request, Success, Failure>) {
        self.operation = operation
        self.childTask = ChildTask(parent: operation)
        resultToken = operation.subscribe(resultListener: resultListener)
    }
```

Then all of the stored properties which reference these types are defined for that instance.

```swift
class AmplifyOperationTaskAdapter<Request: AmplifyOperationRequest, Success, Failure: AmplifyError>: AmplifyTask {
    let operation: AmplifyOperation<Request, Success, Failure>
    let childTask: ChildTask<Void, Success, Failure>
```

Then `AmplifyTask` has the associated types defined for the protocol.

```swift
public protocol AmplifyTask {
    associatedtype Request
    associatedtype Success
    associatedtype Failure: AmplifyError
```

This typealias allows the one outside this type to inherit those type definitions so they do not have to be repeated. Below you can see it maps `FastOperation` to `FastTask` using the nested `TaskAdapter` definition.

```swift
public typealias FastTask = FastOperation.TaskAdapter
```

Without the nested typealias it would be necessary to repeat the list of generic types which is redundant and potentially easy to cause a mismatch. Nesting prevents these concerns.

Reply:
```suggestion
    public typealias TaskAdapter = AmplifyOperationTaskAdapter <Request, Success, Failure>
```